library/src/main/java/com/google/android/exoplayer2/metadata/id3/Id3Decoder.java

```diff
@@ -444,6 +450,112 @@ private static TextInformationFrame decodeTextInformationFrame(ParsableByteArray
     return new TextInformationFrame(id, description);
   }
 
+  private static UrlLinkFrame decodeUrlLinkFrame(ParsableByteArray id3Data,
+      int frameSize) throws UnsupportedEncodingException {
+    int encoding = id3Data.readUnsignedByte();
+    String charset = getCharsetName(encoding);
+
+    byte[] data = new byte[frameSize - 1];
+    id3Data.readBytes(data, 0, frameSize - 1);
+
+    int descriptionEndIndex = indexOfEos(data, 0, encoding);
+    String description = new String(data, 0, descriptionEndIndex, charset);
+
+    int urlStartIndex = descriptionEndIndex + 1;
+    int urlEndIndex = indexOfEos(data, urlStartIndex, encoding);
+    int urlLength = urlEndIndex - urlStartIndex;
+    String url = new String(data, urlStartIndex, urlLength, charset);
+
+    return new UrlLinkFrame(description, url);
+  }
+
+  private static ChapterFrame decodeChapterFrame(ParsableByteArray id3Data, int frameSize,
+      int majorVersion, boolean unsignedIntFrameSizeHack) throws UnsupportedEncodingException {
+    byte[] frameBytes = new byte[frameSize];
+    id3Data.readBytes(frameBytes, 0, frameSize - 1);
+
+    ParsableByteArray chapterData = new ParsableByteArray(frameBytes);
+
+    int chapterIdEndIndex = indexOfZeroByte(frameBytes, 0) + 1;
+    String chapterId = chapterData.readNullTerminatedString(chapterIdEndIndex);
+
+    chapterData.setPosition(chapterIdEndIndex);
+    int startTime = chapterData.readInt();
+    int endTime = chapterData.readInt();
+    int startOffset = chapterData.readInt();
+    int endOffset = chapterData.readInt();
+
+    String title = null;
+    String url = null;
+    ApicFrame image = null;
+
+    int frameHeaderSize = majorVersion == 2 ? 6 : 10;
+    while (chapterData.bytesLeft() >= frameHeaderSize) {
+      Id3Frame frame = decodeFrame(majorVersion, chapterData, unsignedIntFrameSizeHack);
+      if (frame == null) {
+        continue;
+      }
+      if (frame instanceof TextInformationFrame) {
+        TextInformationFrame textFrame = (TextInformationFrame)frame;
+        if (textFrame.id != null && textFrame.id.equals("TIT2")) {
```

Comment:
By convention we do CONST.equals(variable) because it cannot result in a null pointer exception. It also simplifies the expression here, since you can just do:
```
"TIT2".equals(textFrame.id)
```